src/weapons.cpp

```diff
@@ -460,6 +485,19 @@ uint32_t Weapon::getManaCost(const Player* player) const
 	return (player->getMaxMana() * manaPercent) / 100;
 }
 
+int32_t Weapon::getHealthCost(const Player* player) const
```

Comment:
It's basicly kinda custom but I've seen this requested for almost a decade now, so while re writing this stuff I've just picked it up aswell, as it was just a minor thing todo.
It enables that you can make weapons which take health from yourself when you use the weapon, so basicly the equivalent to mana.

Well `WeaponType` or `Weapon` in this case would be entirely the same, in the end the only difference would be that `WeaponType` would be the basic outline and `Weapon` could be the final weapon with attributes.
However as I plan to introduce item virtualization it would take the sense away to have both in this case.
I'd rather have `Attributes` later on which can be applied to the virtual item/weapon then.
In the end a `Weapon` is still an `ItemType` and should be reflected as such.